src/vs/code/electron-main/main.ts

```diff
@@ -223,25 +229,29 @@ class CodeMain {
 		return instanceEnvironment;
 	}
 
-	private initServices(environmentMainService: IEnvironmentMainService, policyService: IPolicyService, configurationService: ConfigurationService, stateMainService: StateMainService): Promise<unknown> {
+	private async initServices(environmentMainService: IEnvironmentMainService, userDataProfilesMainService: UserDataProfilesMainService, configurationService: ConfigurationService, stateMainService: StateMainService): Promise<unknown> {
```

Comment:
There is now a new waterfall it seems. We should try to run all of this in parallel as much as possible. Since this is on the critical startup path.

Reply:
Apparently it is needed because the state service need to be intialized to know about profiles

Reply:
If we consider that `main` always use default profile then we can do all inits parallel like before. I will do that.